easybuild/easyconfigs/d/DETONATE/DETONATE-1.11-GCC-12.3.0.eb

```diff
@@ -0,0 +1,60 @@
+easyblock = 'MakeCp'
+
+name = 'DETONATE'
+version = '1.11'
+
+homepage = 'http://deweylab.biostat.wisc.edu/detonate/'
+description = """DETONATE (DE novo TranscriptOme rNa-seq Assembly with or without the Truth Evaluation)
+ consists of two component packages, RSEM-EVAL and REF-EVAL. Both packages are mainly intended to be used
+ to evaluate de novo transcriptome assemblies, although REF-EVAL can be used to compare sets of any kinds
+ of genomic sequences."""
+
+toolchain = {'name': 'GCC', 'version': '12.3.0'}
+
+toolchainopts = {'cstd': 'c++11'}
+
+source_urls = ['http://deweylab.biostat.wisc.edu/detonate/']
+sources = ['detonate-1.11.tar.gz']
+patches = [
+    'DETONATE-%(version)s_fix-deps.patch',
+    'DETONATE-%(version)s_Intel-fixes.patch',
```

Comment:
is the Intel-fixes patch needed when using GCC? And if it is, perhaps it should be renamed?

Reply:
This is an exact copy of the existing `detonate-1.11_Intel-fixes.patch` (which I didn't rename in #19569, I should have).

So rather than copying it, we should rename the existing patch file to use `DETONATE`, and then we might as well also change the `Intel-fixes` if it's also relevant when using GCC (it might well be for more recent GCC versions).

Reply:
Patch is definitely relevant also with recent `GCC`, without it I get:

```
g++ -O2 -ftree-vectorize -march=native -fno-math-errno -std=c++11 -c -I. parseIt.cpp
In file included from parseIt.cpp:22:
SingleHit.h: In member function bool SingleHit::read(std::istream&):
SingleHit.h:46:24: error: cannot convert std::basic_istream<char>::__istream_type {aka std::basic_istream<char>} to bool in return
   46 |         return (in>>sid>>pos);
      |                ~~~~~~~~^~~~~~
      |                        |
      |                        std::basic_istream<char>::__istream_type {aka std::basic_istream<char>}
```

I'll look into the renaming/cleanup effort right now...

Reply:
fixed in ef0cc1c